pkg/kp/pcstore/consul_store.go

```diff
@@ -272,6 +255,62 @@ func (s *consulStore) Watch(id fields.ID, quit <-chan struct{}) <-chan WatchedPo
 	return outCh
 }
 
+// WatchPodCluster implements a watch for the Pod cluster at _id_
+// It will return a blocking channel on which the client can read
+// WatchedPodCluster objects. The goroutine maintaining the watch will block on
+// writing to this channel so it's up to the caller to read it with haste.
+func (s *consulStore) WatchPodCluster(id fields.ID, quit <-chan struct{}) <-chan WatchedPodCluster {
+	inCh := make(chan *api.KVPair)
+	outCh := make(chan WatchedPodCluster)
+	errChan := make(chan error, 1)
+	key := path.Join(podClusterTree, string(id))
+
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			case err := <-errChan:
+				s.logger.WithError(err).Errorf("WatchPrefix returned error, recovered.")
+			}
+		}
+	}()
+
+	go consulutil.WatchSingle(key, s.kv, inCh, quit, errChan)
+
+	go func() {
+		var kvp *api.KVPair
+		for {
+			select {
+			case kvp = <-inCh:
+				if kvp == nil {
+					// PodCluster at id has been deleted
```

Comment:
shall we do something to let the caller know? for example return `WatchedPodCluster{Err: NoPodCluster}`?

Then again, `rcstore` does not do that either, it just does `continue`... which is different from `return` in that I believe it will wait for the pod cluster to reappear, though I'm not sure how probable that is either. Will give it a little thought.